python/triton/runtime/autotuner.py

```diff
@@ -211,14 +213,18 @@ def run(self, *args, **kwargs):
         self.nargs = None
         return ret
 
-    def prune_configs(self, kwargs):
+    def prune_configs(self, kwargs: Dict) -> List[Config]:
         pruned_configs = self.configs
         if self.early_config_prune:
             pruned_configs = self.early_config_prune(self.configs, self.nargs, **kwargs)
         if self.perf_model:
             top_k = self.configs_top_k
             if isinstance(top_k, float) and top_k <= 1.0:
                 top_k = int(len(self.configs) * top_k)
+            else:
+                # Slice index must be an integer
+                # if top_k is a float > 1.0, we need to cast to int or we have a type error
+                top_k = int(top_k)
```

Comment:
Perhaps we should just raise an error if `top_k` is a float greater than 1? It's strange to have a value-dependent interpretation.

Reply:
yeah we can raise an exception here instead, probably not valueexception (just a placeholder for now)